modulemd/v2/modulemd-component-module.c

```diff
@@ -172,6 +172,34 @@ modulemd_component_module_set_property (GObject *object,
 }
 
 
+gboolean
+modulemd_component_module_equals (ModulemdComponentModule *self,
```

Comment:
This function needs to chain up to the `klass->equals()` function of the parent, or it will not be checking buildorder, name or rationale.

See [`modulemd_component_module_copy()`](https://github.com/fedora-modularity/libmodulemd/blob/9d4d6f5a7a1f45c94062117ade0e30c665bef052/modulemd/v2/modulemd-component-module.c#L67) for an example of how to do this.

Reply:
I am having trouble doing this. This is how I am chaining up, currently:

ModulemdComponentModuleClass *klass = MODULEMD_COMPONENT_MODULE (
204           MODULEMD_COMPONENT_CLASS (modulemd_component_module_parent_class)
205             ->equals (self, other));
206 
207   if (!klass->equals (self, other))
208       return FALSE;

... This results in the following warning and error:

../modulemd/v2/modulemd-component-module.c: In function ‘modulemd_component_module_equals’:
../modulemd/v2/modulemd-component-module.c:204:11: warning: passing argument 1 of ‘MODULEMD_COMPONENT_MODULE’ makes pointer from integer without a cast [-Wint-conversion]
           MODULEMD_COMPONENT_CLASS (modulemd_component_module_parent_class)
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
             ->equals (self, other));
             ~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from ../modulemd/v2/include/modulemd-2.0/modulemd-component-module.h:16,
                 from ../modulemd/v2/modulemd-component-module.c:14:
/usr/include/glib-2.0/gobject/gtype.h:1405:63: note: expected ‘gpointer’ {aka ‘void *’} but argument is of type ‘gboolean’ {aka ‘int’}
   static inline ModuleObjName * MODULE##_##OBJ_NAME (gpointer ptr) {                                     \
                                                      ~~~~~~~~~^~~
../modulemd/v2/include/modulemd-2.0/modulemd-component-module.h:29:1: note: in expansion of macro ‘G_DECLARE_FINAL_TYPE’
 G_DECLARE_FINAL_TYPE (ModulemdComponentModule,
 ^~~~~~~~~~~~~~~~~~~~
../modulemd/v2/modulemd-component-module.c:203:41: warning: initialization of ‘ModulemdComponentModuleClass *’ {aka ‘struct <anonymous> *’} from incompatible pointer type ‘ModulemdComponentModule *’ {aka ‘struct _ModulemdComponentModule *’} [-Wincompatible-pointer-types]
   ModulemdComponentModuleClass *klass = MODULEMD_COMPONENT_MODULE (
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~
../modulemd/v2/modulemd-component-module.c:207:13: error: ‘ModulemdComponentModuleClass’ {aka ‘struct <anonymous>’} has no member named ‘equals’
   if (!klass->equals (self, other))

Reply:
I apologize if the formatting of the above snippets is less than ideal. If there is any way I can improve the formatting, please let me know.